app/views/users/show.html.erb

```diff
@@ -20,8 +20,13 @@
 <% unless current_user_role? == "Student" %>
   <hr/>
   <p><%= link_to 'Edit', :action => 'edit', :id => @user %> |
-    <%= link_to 'Delete', {:action => 'destroy', :id => @user},
-    :confirm => 'Delete this user?', :method => :post %>
+<!--This user cannot be deleted directly-->
+    <% if (@assignment_participant_num != 0)%> 
```

Comment:
Still need modify this if condition. We want to use the result of ```Response_mapping``` database, but we find it a little werid: we create a new user and do nothing, we can find corresponding recode in ```Response_mapping``` database as reviewer.